src/braket/circuits/free_parameter.py

```diff
@@ -0,0 +1,112 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from numbers import Number
+
+from sympy import Symbol
+
+
+class FreeParameter:
+    """
+    Class 'FreeParameter'
+    """
+
+    def __init__(self, name: str):
+        """
+        Initializes a new :class:'FreeParameter' object.
+        Free parameters can be used in parameterized circuits.
+
+        Args:
+            name: Name of the :class:'FreeParameter'. Can be a unicode value.
+
+        Examples:
+            >>> param1 = FreeParameter("theta")
+            >>> param1 = FreeParameter("\u03B8")
+        """
+        parameter_symbol = Symbol(name)
+        self._name = name
+        self._symbol = parameter_symbol
+        self._parameter_expression = parameter_symbol
+        self._parameter_value = Number
+
+    @property
+    def name(self) -> str:
+        """
+        Returns:
+            str: Name of this parameter.
+        """
+        return self._name
+
+    @property
+    def parameter_value(self) -> Number:
+        """
+        This is set to None until a value is assigned.
+
+        Returns:
+            Number: The value assigned to a parameter.
+        """
+        return self._parameter_value
+
+    @property
+    def symbol(self) -> Symbol:
+        """
+        Returns:
+            Symbol: The symbol for a parameter.
+        """
+        return self._symbol
+
+    def fix_value(self, param_value: Number) -> None:
```

Comment:
Would it make sense just to assign the symbol instead? Otherwise it doesn't look like you're using SymPy functionality at all.

Reply:
The symbol would be used in conjunction with the expression for future use cases similar to the physics use cases captured outside of this PR. Also, later support for multiple values would be beneficial to retain this.

Reply:
I actually thought the consensus was to make FreeParameter a wrapper around SymPy, but as it stands it doesn't look like you're using SymPy at all - could have just used a string for a name and a float for a value instead.

Reply:
I generally agree, if Freeparameter is a wrapper around sympy, we should make use of sympy.

My question is, what happens to the symbol after it's assigned? Does it stick around or is it replaced wholesale? Also, how does this fit into the idea that fixing parameters in a circuit creates a new circuit with fewer FreeParameter objects? Finally, how does this handle parallelism, eg. if we want to run the circuit with multiple values at once, say for a gradient calculation?